Packages/nadena.dev.modular-avatar/Runtime/ModularAvatarObjectRemover.cs

```diff
@@ -0,0 +1,52 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+namespace nadena.dev.modular_avatar.core
+{
+    [ExecuteInEditMode]
+    [DisallowMultipleComponent]
+    [AddComponentMenu("Modular Avatar/MA Object Remover")]
+    public class ModularAvatarObjectRemover : AvatarTagComponent
+    {
+        public bool hideInHierarchy;
+
+        public ModularAvatarObjectRemover()
+        {
+            EditorApplication.delayCall += Refresh;
+        }
+
+        private void OnEnable()
+        {
+            Refresh();
+        }
+
+        private void Refresh()
+        {
+            if (Application.isPlaying) return;
+            gameObject.SetActive(false);
+            if (hideInHierarchy) Hide();
+            // components can be always restored from prefabs, so removing them now does not break anything and will
+            // provide better compatibility with other tools
+            foreach (var componentsInChild in gameObject.GetComponentsInChildren<MeshRenderer>())
+            {
+                if (PrefabUtility.IsPartOfAnyPrefab(componentsInChild)) DestroyImmediate(componentsInChild);
+            }
+            foreach (var componentsInChild in gameObject.GetComponentsInChildren<SkinnedMeshRenderer>())
+            {
+                if (PrefabUtility.IsPartOfAnyPrefab(componentsInChild)) DestroyImmediate(componentsInChild);
+            }
```

Comment:
This feels like surprising behavior from a user perspective - when I hide an object and restore it later, it's missing a bunch of MeshRenderers, and any prefab overrides are lost.

Reply:
Would it be better to just set EditorOnly (in addition to hiding from view?)

Reply:
Well my assumption is that this component is like removing but sadly we just can't remove objects without breaking a prefab. The component is technically still there as removing component from a prefab leaves you with marker that it got removed
While I do prefer this behaviour, I'm fine to just remove it, just some tools then keep complaining about these mesh renderes or want to use them for their actions. 
From my manual testing removing component that is part of a prefab does NOT remove any data, even the changes you did inside the component are still preserved: 
![image](https://user-images.githubusercontent.com/6942692/235183682-9e96b958-86cd-405e-84f2-5c91b72f1430.png)
![image](https://user-images.githubusercontent.com/6942692/235184008-e29de4c5-1549-46c5-bcca-81e8d0247c09.png)
And its back to:
![image](https://user-images.githubusercontent.com/6942692/235184069-9240462c-6a48-4c22-bc0b-c92c939a23d1.png)

Reply:
Added option to restore and made this behaviour use a separate toggle